rdtools/normalization.py

```diff
@@ -513,37 +517,44 @@ def energy_from_power(power, target_frequency=None, max_timedelta=None):
         else:
             raise
 
-    # Upsampling case
-    if freq_interval_size_ns <= median_step_ns:
-        resampled = interpolate(power, target_frequency, max_timedelta)
-
-        moving_average = (resampled + resampled.shift()) / 2.0
-
-        energy = moving_average * t_step_nanoseconds(moving_average) \
-                    / 10.0**9 / 3600.0
-
-        # Drop first row with work around for pandas issue #18031
-        if energy.index.tz is None:
-            energy = energy.drop(energy.index[0])
-        else:
-            tz = str(energy.index.tz)
-            energy.index = energy.index.tz_convert('UTC')
-            energy = energy.drop(energy.index[0])
-            energy.index = energy.index.tz_convert(tz)
+    if len(power) <= 1:
+        # just one value, doesn't make sense to interpolate or trapz aggregate.
+        energy = power * freq_interval_size_ns / 1e9 / 3600  # ns to s to h
```

Comment:
Is `max_timedelta` meaningful for single-value inputs?  I don't think so, so it's not used in this branch.

Reply:
Also we get the timezone and frequency automatically from the input by doing this, so no need to do anything explicit like the main branch does.

Reply:
I agree on both counts